.github/workflows/make-ci.yaml

```diff
@@ -0,0 +1,38 @@
+---
+name: Make CI
+
+'on':
+  workflow_call:
+    inputs:
+      timeout_minutes:
+        description: The maximum time (in minutes) for a job to run.
+        default: 10
+        required: false
+        type: number
+
+jobs:
+  make-ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22.x"
+      - name: Install prettier
+        run: yarn add prettier@3
+      - name: Put node executables in the system PATH
+        run: echo -n "$(yarn bin)" >> $GITHUB_PATH
+      - name: Install poetry
+        run: pipx install poetry
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          cache: poetry
+          python-version: "3.12"
+      - name: Install package
+        run: poetry install
```

Comment:
This definitely needs something like https://github.com/broadinstitute/shared-workflows/blob/main/.github/workflows/python-lint.yaml#L17-L21 for projects where one does not want the install to include the project in the current directory.

Reply:
discussed on meet, no usecase